hypervisor/virt/hyperv/hypervcli.py

```diff
@@ -0,0 +1,104 @@
+import json
+import re
+
+from hypervisor import logger
+from hypervisor.ssh import SSHConnect
+
+
+class HypervCLI:
+    def __init__(self, server, ssh_user, ssh_passwd):
+        """
+        Collect hyperv information, provide Guest add/delete/start/stop/suspend/resume functions
+        via PowerCli Command line
+        :param server: the ip for the hyperv server
+        :param ssh_user: the ssh user for the hyperv server
+        :param ssh_passwd: the ssh password for the hyperv server
+        """
+        self.server = server
+        self.ssh_user = ssh_user
+        self.ssh_passwd = ssh_passwd
+        self.ssh = SSHConnect(self.server, user=self.ssh_user, pwd=self.ssh_passwd)
+
+    def _format(self, ret=0, stdout=None):
+        """
+        Convert the json string to python list data
+        :param ret: return code
+        :param stdout: output for the execute command
+        :return: the list after json.loads
+        """
+        stdout = re.findall(r'[[][\W\w]+[]]', stdout)[0]
+        if ret is 0 and stdout is not None:
+            return json.loads(stdout)
+
+    def info(self):
+        """
+        Get the VMhost info
+        :return: the VMHosts info
+        """
+        cmd = 'PowerShell ConvertTo-Json @(Get-VMHost)'
+        ret, output = self.ssh.runcmd(cmd)
+        output = self._format(ret, output)
+        for host in output:
+            logger.info(f"Get Hyperv master: {host['Name']}")
+        return output
+
+    def guest_search(self, guest_name, uuid_info=False):
+        """
+        Search the specific guest, return the expected attributes
+        :param guest_name: name for the specific guest
+        :param uuid_info: if you need the uuid_info: guest_uuid, hyperv_uuid, hyperv_hwuuid
+        :return: guest attributes, exclude guest_name, guest_ip, guest_uuid ...
+                 guest_state: guest_poweron:1, guest_poweroff:0, guest_Suspended:2
+                 hyperv_state: host_poweron:1, host_poweroff:0
+        """
+        cmd = f'PowerShell ConvertTo-Json @(Get-VM {guest_name})'
+        ret, output = self.ssh.runcmd(cmd)
+        output = self._format(ret, output)[0]
+        guest_msgs = {
+            'guest_name': output['VMName'],
+            'guest_ip': self.guest_ip(guest_name),
+            'guest_state': output['State'],
+            'hyperv_ip': self.server,
+            'hyperv_hostname': output['ComputerName'],
+            'hyperv_version': output['Version'],
+            'hyperv_cpu': output['CPUUsage'],
+        }
+        if uuid_info:
+            guest_msgs['guest_uuid'] = self.guest_uuid()
+            guest_msgs['hyperv_uuid'] = self.host_uuid()
+        return guest_msgs
```

Comment:
The uuid_info is important for virtwho-test project, so I suggest cancel the ```uuid_info=False``` para, return all keys as default.

Reply:
This part is the same as https://github.com/VirtwhoQE/hypervisor-builder/blob/master/hypervisor/virt/esx/powercli.py#L179
The original reason to set the value is that: if you want to get the `guest_state`, you should call `self.guest_search(guest_name)['guest_state']`, and it will execute the get uuid and host uuid and hwuuid commands, it will be executed frequently and time consuming. 
Noted that other values can be got from the output dicts in esx_powercli. But the guest_ip from the hyperv_powercli also need execute other command, seems I cannot find another good way.

So, as above, we can set it `True` as default, but maybe need it.

Reply:
Yeah, understand, let's keep it as it is now and change when have further requirement.